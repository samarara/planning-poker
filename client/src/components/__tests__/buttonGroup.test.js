import TestRenderer from "react-test-renderer"; // ES6
import { render, fireEvent, waitFor, screen } from "@testing-library/react";
import ButtonGroup from "../buttonGroup";

describe("button group", () => {
  it("should match snapshot", () => {
    const props = {
      onCreateRoomClick: jest.fn(),
      onJoinRoomClick: jest.fn(),
      onChange: jest.fn(),
      input: "test input",
      error: false,
      errorMessage: "test error",
    };
    const buttonGroup = TestRenderer.create(<ButtonGroup {...props} />);
    expect(buttonGroup).toMatchSnapshot();
  });

  it("should fire create new room handler", () => {
    const onCreateRoomMock = jest.fn();
    render(<ButtonGroup onCreateRoomClick={onCreateRoomMock} />);
    fireEvent.click(screen.getByText("Create new voting room"));
    expect(onCreateRoomMock).toBeCalled();
  });

  it("should fire join existing room handler", () => {
    const onJoinRoomMock = jest.fn();
    render(<ButtonGroup onJoinRoomClick={onJoinRoomMock} />);
    fireEvent.click(screen.getByText("Join existing room"));
    expect(onJoinRoomMock).toBeCalled();
  });

  it.skip("should fire on change event handler", async () => {
    const onChangeMock = jest.fn();
    render(<ButtonGroup onChange={onChangeMock} />);
    fireEvent.change(await screen.findByText("Enter room code"));
    // console.log(await screen.findByText('Enter room code'));
    expect(onChangeMock).toBeCalled();
  });

  it("should display error message if error prop is true", () => {
    const props = {
      onCreateRoomClick: jest.fn(),
      onJoinRoomClick: jest.fn(),
      onChange: jest.fn(),
      input: "",
      error: true,
      errorMessage: "test error",
    };
    render(<ButtonGroup {...props} />);
    fireEvent.click(screen.getByText("Join existing room"));
    expect(screen.getByText("test error")).toHaveTextContent("test error");
  });
});
