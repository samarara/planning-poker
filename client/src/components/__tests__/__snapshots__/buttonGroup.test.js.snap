// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button group should match snapshot 1`] = `
<div
  className="makeStyles-buttonGroup-1"
  id="button group"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-firstButton-3 MuiButton-containedPrimary MuiButton-fullWidth"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Create new voting room
    </span>
  </button>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
      data-shrink={false}
    >
      Enter room code
      <span
        aria-hidden={true}
        className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-6"
        >
          <span>
            Enter room code
             *
          </span>
        </legend>
      </fieldset>
    </div>
    
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-2 MuiButton-fullWidth"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Join existing room
    </span>
  </button>
</div>
`;
